### DISCUSSION TITLE
Custom sigmoid for mean prior

### DISCUSSION THREAD

**Discussion by CecileLeSueur** [cornellius-gp/gpytorch#2179] [Q&A]
Hello, 

I'm wondering how I could define a sigmoid function as prior for the mean? 

Especially, I'm interested in implementing the following function :
S(x) = p + [ (1-p) / ( 1 + exp( b * (x-a) ) )  ] 

I've been trying several implementations, but each time, the estimated parameters of the sigmoid are such that the sigmoid is almost flat, i.e. it tends towards the constant mean (I've tried with the exact GP regression and with the VI).

Any ideas why?  Do you think it might be linked to the optimisation algorithm which favours a constant mean? If yes, how could I fit the optimal sigmoid simultaneously than the covariance hyperparameters?
Could you also please help me writing the definition of this sigmoid function, this would be very helpful? :)

Thanks a lot for the help!! :)

**Comment by gpleiss** [comment]
> I've been trying several implementations, but each time, the estimated parameters of the sigmoid are such that the sigmoid is almost flat, i.e. it tends towards the constant mean (I've tried with the exact GP regression and with the VI).

This probably depends a lot on your data. In general, I personally think it's best to avoid custom prior means unless you are sure that your data very closely follows a particular functional form.

If your data does indeed look very sigmoidal, you should try initializing the `p`, `b`, and `a` parameters to sensible defaults that match your data well.
